packages/core/src/components/dialog/_dialog.scss

```diff
@@ -35,34 +38,48 @@ Styleguide pt-dialog
 */
 
 $dialog-border-radius: $pt-border-radius * 2 !default;
+$dialog-margin: ($pt-grid-size * 3) 0 !default;
 $dialog-padding: $pt-grid-size * 2 !default;
 
-.pt-dialog {
+.pt-dialog-container {
   $dialog-transition-props: (
     opacity: (0, 1),
-    transform: (translateX(50%) scale(0.5), translateX(50%) scale(1))
+    transform: (scale(0.5), scale(1))
   );
 
   @include react-transition(
     "pt-overlay",
     $dialog-transition-props,
     $duration: $pt-transition-duration * 3,
     $easing: $pt-transition-ease-bounce,
-    $before: "&"
+    $before: "&",
+    $after: "> .pt-dialog"
   );
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  width: 100%;
+  min-height: 100%;
+  user-select: none;
+
+  // LEGACY: override old (<= 1.24.0) dialog styles when inside a container to respect flex layout
+  .pt-dialog {
+    // stylelint-disable-next-line declaration-no-important
+    position: static !important;
```

Comment:
Hey can this be reasonably removed? This is some pretty wicked styling to overcome if you don't want your dialog to be in the center of the screen.

Reply:
@kmwhelan93 please file a new issue to discuss this. nearly lost this comment to the sands of time. also note that we're working on 3.0 on `develop`, while this PR is way back in the 1.x days.

Reply:
@kmwhelan93 or even better, hit me with a PR and a passing build! ✅